controllers/cluster/cluster_controller.go

```diff
@@ -113,8 +113,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	probeConfig := r.getEffectiveProbeConfig(shoot, log)
 	if canStartProber(shoot) {
-		r.startProber(ctx, log, req.Name)
+		r.startProber(ctx, probeConfig, log, req.Name)
```

Comment:
(thanks for PR enhacements). There is already a `probeConfig` in the reconciler ? Why not override it after computing the effective `probeConfig`. Otherwise there is a stale `probeConfig` left dangling.

Reply:
The `probeConfig` of the reconciler is the default probe config generated from the config map used by DWD. We generate an effectiveProbeConfig because the `NodeMonitorGracePeriod` of KCM which is used in DWD as well (`kcmNodeMonitorGraceDuration`) is configurable via the shoot yaml. We should not change the config in the reconciler because otherwise the default value for `kcmNodeMonitorGraceDuration` will be lost.

Reply:
To add what @rishabh-11 has already stated, probeConfig is one per seed and shared by all probers. However there is shoot specific configuration that we now need to consider which can override the nodeMonitorGracePeriod for a specific shoot and therefore a specific probe. Other shoots and their corresponding probes will continue to use the seed level prober config and only shoots where `nodeMonitorGracePeriod` has been explicitly set will take that value and thus have a different probe config.

Reply:
Got it. Maybe we could call the field in the reconciler struct `baseProbeConfig` or `defaultProbeConfig`

Reply:
done